### ISSUE TITLE
Failed to commit a transaction with a large insert stmt

### ISSUE THREAD

**Issue by zyguan** [pingcap/tidb#15073]
Please answer these questions before submitting your issue. Thanks!

1. What did you do?

Add the following config to enable large transaction support:
```toml
[performance]
txn-total-size-limit = 10737418240
```
Then create two tables with same schema, insert 3G data to `sbtest1`, and finally execute  `insert into sbtest2 select * from sbtest1`:
```sql
create table if not exists sbtest1 (id int(11) not null primary key, k int(11) not null, c char(120) not null default '', pad char(255) not null default '');
create table if not exists sbtest2 (id int(11) not null primary key, k int(11) not null, c char(120) not null default '', pad char(255) not null default '')
-- insert 3G data to sbtest1
insert into sbtest2 select * from sbtest1;
```

2. What did you expect to see?

(Maybe) the insertition should succeed.

3. What did you see instead?

The insert stmt failed with `Error 1467: Failed to read auto-increment value from storage engine`, due to the [`maxStep`](https://github.com/pingcap/tidb/blob/0bbf1d954c85a2dc6010f4d1f412e405f2b14b02/meta/autoid/autoid.go#L52) limit ([here](https://github.com/pingcap/tidb/blob/0bbf1d954c85a2dc6010f4d1f412e405f2b14b02/meta/autoid/autoid.go#L596) the `tmpStep` won't be greater than `maxStep`).

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

```
> select tidb_version();
+-------------------------------------------------------------------+
| tidb_version()                                                    |
+-------------------------------------------------------------------+
| Release Version: v4.0.0-beta-255-gded862fbe-dirty                 |
| Git Commit Hash: ded862fbebc555de98e230ef57310f9162725a9e         |
| Git Branch: master                                                |
| UTC Build Time: 2020-03-02 12:48:26                               |
| GoVersion: go1.13.7                                               |
| Race Enabled: false                                               |
| TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306 |
| Check Table Before Drop: false                                    |
+-------------------------------------------------------------------+
```

**Comment by zyguan** [comment]
@tiancaiamao PTAL

**Comment by jackysp** [comment]
@AilinKid PTAL

**Comment by AilinKid** [comment]
@Little-Wallace  here will be a bug leading autoid exhausted quickly. last time we found it in `load data`, here in insert select from. PTAL

**Comment by zyguan** [comment]
The issue has been fixed by #15261 , but since #15251 set oom-action to cancel by default, we need to mention it in document of large transaction.
